### PR TITLE
Task/add date time utils

### PR DESCRIPTION
**Pull request by IvanHdzC**
This PR modifies project to use maven modules.

This will make easier to include new libraries into perseo-core project as new modules and include them into perseo-core as dependencies.

All perseo-core code files were refactored into perseo-main module. perseo-core now is used as parent module to package perseo-main and perseo-utils as child projects.

Finally on docker file war file is deployed as perseo-core for any remaining name dependency.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,16 +24,16 @@ WORKDIR /code
```
Comment (fgalan) [inline Dockerfile]: I understand perseo-main/.DS_Store should not be included in this PR. Maybe it should be added to .gitignore?

File: perseo-main/pom.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (fgalan) [inline perseo-main/pom.xml]: Licences header in all new files, please (copy paste from a similar existing one).

File: perseo-main/src/main/java/com/telefonica/iot/perseo/Utils.java [hunk 3]
```diff
@@ -76,12 +78,31 @@ public static synchronized EPServiceProvider initEPService(ServletContext sc) {
             cfg.addEventType(Constants.IOT_EVENT, def);
 
             // Add SunriseSunset library
-            cfg.addImport("ca.rmen.sunrisesunset.*");
+            //cfg.addImport("ca.rmen.sunrisesunset.*");
```
Comment (fgalan) [inline perseo-main/src/main/java/com/telefonica/iot/perseo/Utils.java]: Leftover comment?

File: perseo-main/src/main/java/com/telefonica/iot/perseo/Utils.java [hunk 4]
```diff
@@ -76,12 +78,31 @@ public static synchronized EPServiceProvider initEPService(ServletContext sc) {
...
+
+            // Add SunriseSunset library
+            cfg.addImport("com.telefonica.iot.perseo.utils.*");
+
             // Add Single row function for getSunriseSunset
             try {
-                cfg.addPlugInSingleRowFunction("getSunriseSunset",
-                                               "ca.rmen.sunrisesunset.SunriseSunset",
-                                               "getSunriseSunset");
+                //cfg.addPlugInSingleRowFunction("getSunriseSunset",
```
Comment (fgalan) [inline perseo-main/src/main/java/com/telefonica/iot/perseo/Utils.java]: Leftover comment?

File: perseo-utils/src/main/java/com/telefonica/iot/perseo/utils/DateTimeUtils.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+import java.util.Calendar;
+
+public class DateTimeUtils {
+
+    public static Calendar getNextSunise(Calendar day, double latitude, double longitude) {
```
Comment (fgalan) [inline perseo-utils/src/main/java/com/telefonica/iot/perseo/utils/DateTimeUtils.java]: Not sure about the JavaDoc policy in this project... In Cygnus I think we use JavaDoc in the code, but not sure in Perseo Core. JavaDoc is always welcome :)

File: perseo-utils/src/main/java/com/telefonica/iot/perseo/utils/DateTimeUtils.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+import java.util.Calendar;
+
+public class DateTimeUtils {
```
Comment (fgalan) [inline perseo-utils/src/main/java/com/telefonica/iot/perseo/utils/DateTimeUtils.java]: Are the methods included in this class properly covered by unit tests?

**Comment by fgalan** [comment]
This PR should add an entry (or sereval) about the changes in CHANGEX_NEXT_RELEASE file. We can discuss the exact wording when we meet @IvanHdzC

File: Dockerfile (hunk 1 above)
Comment (IvanHdzC) [reply Dockerfile]: Fixed in [ccd0caf](https://github.com/telefonicaid/perseo-core/pull/143/commits/ccd0caff61eda1b5cc612365e9d02e766f3fd21e)

File: perseo-main/pom.xml (hunk 2 above)
Comment (IvanHdzC) [reply perseo-main/pom.xml]: Fixed in [9396d00](https://github.com/telefonicaid/perseo-core/pull/143/commits/9396d0097ab22859d06b72f3f02772efb1065699)

File: perseo-main/src/main/java/com/telefonica/iot/perseo/Utils.java (hunk 3 above)
Comment (IvanHdzC) [reply perseo-main/src/main/java/com/telefonica/iot/perseo/Utils.java]: Fixed in [f3f840b](https://github.com/telefonicaid/perseo-core/pull/143/commits/f3f840b4bcf70c19c14266beabc9f568ec1feec1)

File: perseo-main/src/main/java/com/telefonica/iot/perseo/Utils.java (hunk 4 above)
Comment (IvanHdzC) [reply perseo-main/src/main/java/com/telefonica/iot/perseo/Utils.java]: Fixed in [f3f840b](https://github.com/telefonicaid/perseo-core/pull/143/commits/f3f840b4bcf70c19c14266beabc9f568ec1feec1)

File: perseo-utils/src/main/java/com/telefonica/iot/perseo/utils/DateTimeUtils.java (hunk 5 above)
Comment (IvanHdzC) [reply perseo-utils/src/main/java/com/telefonica/iot/perseo/utils/DateTimeUtils.java]: Fixed in [0302210](https://github.com/telefonicaid/perseo-core/pull/143/commits/03022106ab35fdd3cd7aec7d76e423a429de54aa)

File: perseo-utils/src/main/java/com/telefonica/iot/perseo/utils/DateTimeUtils.java (hunk 6 above)
Comment (IvanHdzC) [reply perseo-utils/src/main/java/com/telefonica/iot/perseo/utils/DateTimeUtils.java]: Fixed in [f01e6a3](https://github.com/telefonicaid/perseo-core/pull/143/commits/f01e6a34ebc65a7ca3cb90e29f4bcadcc1a4e4a3)

**Comment by IvanHdzC** [comment]
> This PR should add an entry (or sereval) about the changes in CHANGEX_NEXT_RELEASE file. We can discuss the exact wording when we meet @IvanHdzC

Fixed in [4c1b194](https://github.com/telefonicaid/perseo-core/commit/4c1b1949e3608f6d01ed6e5359cef8a3a3499acd)
